File: backend/MovieApi/Models/Movie.cs
namespace MovieApi.Models;

public class Movie
{
    public int Id { get; set; }
    public string Title { get; set; }
    public int  Year { get; set; }
    public string Genre { get; set; }
    public int Duration { get; set; }
    
    //1:1
    public MovieDetails MovieDetails { get; set; }
    //1:M
    public List<Review> Reviews { get; set; }
    //N:M
    public List<MovieActor> MovieActors { get; set; }
}

File: frontend/src/types/movie.ts
export interface Movie  {
    id: number;
    title: string;
    year: number;
    genre: string;
    duration: number;

}

export interface MovieCreateDto {
    title: string;
    year: number;
    genre: string;
    duration: number;
}

export interface Actor {
    id: number;
    name: string;
    birthYear: number;
    role: string;
}

export interface Review {
    id:  number;
    reviewerName: string;
    comment: string;
    rating: number;
}

export interface ReviewCreateDto {
    reviewerName: string;
    rating: number;
    comment: string;
}

export interface MovieDetails {
    id: number;
    title: string;
    year: number;
    genre: string;
    duration: number;
    synopsis: string;
    language: string;
    actors: Actor[];
    reviews: Review[];
}